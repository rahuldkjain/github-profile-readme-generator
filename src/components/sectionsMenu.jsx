import React, { useState } from 'react';

const SectionsMenu = () => {
  const [isOpen, setIsOpen] = useState(false);

  const sections = [
    { id: 'title-section', label: 'Title' },
    { id: 'subtitle-section', label: 'Subtitle' },
    { id: 'work-section', label: 'Work' },
    { id: 'skills-section', label: 'Skills' },
    { id: 'social-section', label: 'Social' },
    { id: 'addons-section', label: 'Addons' },
    { id: 'support-section', label: 'Support' },
  ];

  const scrollToSection = (sectionId) => {
    const element = document.getElementById(sectionId);
    if (element) {
      element.scrollIntoView({ behavior: 'smooth' });
    }
  };

  return (
    <div className="fixed right-0 top-0 p-4" style={{ zIndex: 999999 }}>
      <button
        onClick={() => setIsOpen(!isOpen)}
        className="text-white rounded-full p-3 shadow-lg transition-all duration-200 focus:outline-none"
        style={{ background: '#f3661aff', transform: 'scale(1)' }}
      >
        {isOpen ? (
          <svg
            xmlns="http://www.w3.org/2000/svg"
            className="h-6 w-6"
            fill="none"
            viewBox="0 0 24 24"
            stroke="currentColor"
          >
            <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M6 18L18 6M6 6l12 12" />
          </svg>
        ) : (
          <svg
            xmlns="http://www.w3.org/2000/svg"
            className="h-6 w-6"
            fill="none"
            viewBox="0 0 24 24"
            stroke="currentColor"
          >
            <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M4 6h16M4 12h16M4 18h16" />
          </svg>
        )}
      </button>

      {isOpen && (
        <div className="absolute right-0 mt-2 mx-2 w-48 bg-white rounded-lg shadow-xl border border-gray-200">
          <div className="px-4 py-2 border-b border-gray-200 font-medium text-gray-700">Sections</div>
          <ul className="py-2">
            {sections.map((section) => (
              <li key={section.id}>
                <button
                  onClick={() => scrollToSection(section.id)}
                  className="w-full text-left px-4 py-2 hover:bg-gray-300 transition-colors duration-200 focus:outline-none"
                >
                  {section.label}
                </button>
              </li>
            ))}
          </ul>
        </div>
      )}
    </div>
  );
};

export default SectionsMenu;
