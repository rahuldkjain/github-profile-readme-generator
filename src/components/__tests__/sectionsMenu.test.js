import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SectionsMenu } from '../sectionsMenu';

// Mock categorizedSkills similar to skills.test.js
jest.mock('../../constants/skills', () => ({
  __esModule: true,
  categorizedSkills: {
    language: {
      title: 'Programming Languages',
      skills: ['javascript'],
    },
    frontend_dev: {
      title: 'Frontend Development',
      skills: ['react'],
    },
  },
}));

describe('SectionsMenu', () => {
  let component;

  beforeEach(() => {
    // Mock the document.getElementById and scrollIntoView
    document.getElementById = jest.fn((id) => ({
      scrollIntoView: jest.fn(),
    }));
    component = shallow(<SectionsMenu />);
  });

  it('renders correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('starts with a closed menu', () => {
    expect(component.find('.absolute').exists()).toBeFalsy();
  });

  it('opens menu when clicking the toggle button', () => {
    // Find and click the menu button
    component.find('button').first().simulate('click');
    expect(component.find('.absolute').exists()).toBeTruthy();
  });

  it('closes menu when clicking the toggle button again', () => {
    // Open menu
    component.find('button').first().simulate('click');
    expect(component.find('.absolute').exists()).toBeTruthy();

    // Close menu
    component.find('button').first().simulate('click');
    expect(component.find('.absolute').exists()).toBeFalsy();
  });

  describe('Section navigation', () => {
    beforeEach(() => {
      component.find('button').first().simulate('click');
    });

    it('calls scrollToSection when clicking main section buttons', () => {
      const mockScrollIntoView = jest.fn();
      document.getElementById = jest.fn(() => ({
        scrollIntoView: mockScrollIntoView,
      }));

      // Click the Title section button
      const titleButton = component
        .findWhere((node) => node.type() === 'button' && node.text().includes('Title'))
        .first();
      titleButton.simulate('click');

      expect(document.getElementById).toHaveBeenCalledWith('title-section');
      expect(mockScrollIntoView).toHaveBeenCalledWith({ behavior: 'smooth' });
    });
  });
});
