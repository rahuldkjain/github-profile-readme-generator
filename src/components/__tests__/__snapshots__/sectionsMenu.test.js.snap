// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionsMenu renders correctly 1`] = `
<div
  className="fixed right-0 top-0 p-4"
  style={
    Object {
      "zIndex": 999999,
    }
  }
>
  <button
    className="text-white rounded-full p-3 shadow-lg transition-all duration-200 focus:outline-none"
    onClick={[Function]}
    style={
      Object {
        "background": "#f3661aff",
        "transform": "scale(1)",
      }
    }
  >
    <svg
      className="h-6 w-6"
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 6h16M4 12h16M4 18h16"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
      />
    </svg>
  </button>
</div>
`;
